leakpro/input_handler/modality_extensions/image_extension.py

```diff
@@ -0,0 +1,37 @@
+"""Module for handling image data with preprocessing and normalization."""
+import numpy as np
+import torch
+from PIL import Image
+from torch import Tensor
+from torchvision import transforms
+
+from leakpro.utils.import_helper import Self
+from leakpro.utils.logger import logger
+
+
+class ImageExtension:
+    """Class for handling image data with preprocessing and normalization."""
+
+    def __init__(self: Self, transform: transforms.Compose) -> None:
+        """Initialize the ImageExtension class and perform initial checks."""
+        x, y = next(iter(self.get_dataloader(0)))
+        if not isinstance(x, (Tensor, np.ndarray, Image.Image)) or not isinstance(y, (Tensor, np.ndarray)):
+            raise ValueError("Data must be a tensor, numpy array, or PIL image.")
+
+        if transform is not None:
+            self.transform = transform
+            logger.info("ImageExtension initialized with custom transformations.")
+            return
+
+        logger.info("ImageExtension initialized with NO transformations.")
+
+    def preprocess_image(self: Self, image: Image.Image) -> Tensor:
```

Comment:
Be careful. When reading from a dataloader, if you have a transform in the __get_item__ the image will already be transformed. However, if you read from dataloader.dataset, it will not have the transform applied.